examples/CMakeLists.txt.in

```diff
@@ -14,7 +14,7 @@
 # limitations under the License.
 #===============================================================================
 
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.18.2)
```

Comment:
(random spot)
There are a few places in the build system that we want to adjust after this change:
* https://github.com/intel-innersource/libraries.performance.math.onednn/blob/e3c3d4799afe447f314eb00e4bcc1009833ab195/cmake/SYCL.cmake#L139
* https://github.com/intel-innersource/libraries.performance.math.onednn/blob/e3c3d4799afe447f314eb00e4bcc1009833ab195/cmake/platform.cmake#L316
* https://github.com/intel-innersource/libraries.performance.math.onednn/blob/e3c3d4799afe447f314eb00e4bcc1009833ab195/cmake/utils.cmake#L168
* https://github.com/intel-innersource/libraries.performance.math.onednn/blob/e3c3d4799afe447f314eb00e4bcc1009833ab195/examples/CMakeLists.txt.in#L93
* https://github.com/intel-innersource/libraries.performance.math.onednn/blob/e3c3d4799afe447f314eb00e4bcc1009833ab195/examples/CMakeLists.txt.in#L125